taichi/backends/cc/codegen_cc.cpp

```diff
@@ -381,8 +381,13 @@ class CCTransformer : public IRVisitor {
     const auto type = stmt->element_type();
     auto var = define_var(cc_data_type_name(type), stmt->raw_name());
     emit("{} = *{};", var, dest_ptr);
-    if (stmt->op_type == AtomicOpType::max ||
-        stmt->op_type == AtomicOpType::min) {
+    if (stmt->op_type == AtomicOpType::cas) {
+      TI_ASSERT(stmt->comp);
+      const auto comp_name = stmt->comp->raw_name();
+      emit("if ({} == {})", stmt->raw_name(), comp_name);
```

Comment:
I think the semantics of CAS is: if **dst** == `comp`, then store `val` to `dst`? `stmt` itself should store the old value stored in `dst`.

Reply:
My current:
```cpp
old = *dst;
if (old == comp)
  *dst = val;
```
Your suggestion:
```cpp
old = *dst;
if (*dst == comp)
  *dst = val;
```
They are the same since C backend is always single thread.

Reply:
I kind of agree with @k-ye that this part is confusing. It took me a while to realize that in Ln 383 that variable `stmt->raw_name()` is set to the old value. Also the correctness depends on the fact that the C backend is single-threaded. Some comments here would be helpful.